rest/config.go

```diff
@@ -246,8 +246,9 @@ type DeprecatedCacheConfig struct {
 }
 
 type RevCacheConfig struct {
-	Size       *uint32 `json:"size,omitempty"`        // Maximum number of revisions to store in the revision cache
```

Comment:
We can't remove config options like this, it would be a breaking change for customers with existing configuration.

If we want to keep the `size` JSON field name, and rename the internal struct field to `MaxItemCount` to better represent the behaviour in the codebase, we can do that. The alternatve would be adding a new property, deprecating `Size` but still allowing it as a fallback.

Both of these options come with downsides.